packages/cli/src/rpc/file/update.ts

```diff
@@ -0,0 +1,139 @@
+import { ExplainRpc, FileRpc } from '@appland/rpc';
+import { RpcHandler } from '../rpc';
+import { exists } from '../../utils';
+import { readFile, writeFile } from 'fs/promises';
+import { warn } from 'console';
+import { INavieProvider } from '../explain/navie/inavie';
+import { explain, explainStatus } from '../explain/explain';
+import applyFileUpdate from './applyFileUpdate';
+
+function reportError(operation: string, e: Error | unknown, file: string) {
+  const tokens = [`[file-update] Failed to ${operation} ${file}`];
+  if (e instanceof Error) tokens.push(e.message);
+  warn(tokens.join(': '));
+}
+
+async function createFile(file: string, modified: string): Promise<FileRpc.UpdateResponse> {
+  try {
+    await writeFile(file, modified, 'utf-8');
```

Comment:
If we're creating new files, we should consider `mkdir(dirname(file), { recursive: true })` before writing the file. I've run in to cases where the LLM will add a new directory, and this writeFile will fail.